### PR TITLE
fix: correct Linux distro version in systeminfo prompt

### PR DESCRIPTION
**Pull request by brayo-pip**
> [!IMPORTANT]
> Corrects Linux OS version retrieval in `prompt_systeminfo()` by using `get_system_distro_version()` instead of `platform.uname().release`.
> 
>   - **Behavior**:
>     - Corrects the method to obtain Linux OS version in `prompt_systeminfo()` by replacing `platform.uname().release` with `get_system_distro_version()`.
>   - **Functions**:
>     - Adds `get_system_distro_version()` to extract the Linux distribution version from `/etc/os-release` using regex.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=ErikBjare%2Fgptme&utm_source=github&utm_medium=referral)<sup> for f57e55af42e5bba1998deca47722cde3ca35107a. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

**Comment by 0xbrayo** [comment]
Closes #238

File: gptme/prompts.py [hunk 1]
```diff
@@ -247,6 +247,18 @@ def get_system_distro() -> str:
     return "Linux"
 
 
+def get_system_distro_version() -> str:
+    """Get the system distribution version."""
+    regex = re.compile(r"^BUILD_ID=\"?([^\"]+)\"?")
```
[the remark this answers is not in the document]
Comment (0xbrayo) [reply gptme/prompts.py]: On arch it's BUILD_ID. But even co-pilot suggested the same thing. Let me check on fedora.

File: gptme/prompts.py [hunk 2]
```diff
@@ -247,6 +247,18 @@ def get_system_distro() -> str:
     return "Linux"
 
 
+def get_system_distro_version() -> str:
```
Comment (ErikBjare) [inline gptme/prompts.py]: I would merge this with `get_system_distro` and make it return a `tuple[str, str]`

**Comment by ErikBjare** [comment]
Just learned about `platform.freedesktop_os_release()` which returns the contents of `/etc/os-release` as a dict

File: gptme/prompts.py [hunk 3]
```diff
@@ -215,8 +214,10 @@ def prompt_tools(examples: bool = True) -> Generator[Message, None, None]:
 def prompt_systeminfo() -> Generator[Message, None, None]:
     """Generate the system information prompt."""
     if platform.system() == "Linux":
-        os_info = get_system_distro()
-        os_version = platform.uname().release
+        os_info = platform.freedesktop_os_release().get("NAME", "Linux")
+        os_version = platform.freedesktop_os_release().get(
+            "VERSION_ID", platform.freedesktop_os_release().get("BUILD_ID", "")
+        )
```
Comment (ErikBjare) [inline gptme/prompts.py]:
```suggestion
        release_info = platform.freedesktop_os_release()
        os_info = release_info.get("NAME", "Linux")
        os_version = release_info.get("VERSION_ID") or release_info.get("BUILD_ID", "")
```

**Pull request merged**